File: mars/src/main/java/Rover.java
import java.util.LinkedList;

/**
 * @author guang
 * @since 2019-08-22 14:45
 */
public class Rover {

    private Point point;

    public void deploy(Point point) {
        this.point = point;
    }

    public Point run(Instruction instruction) {
        if (null == point) {
            throw new RuntimeException("请初始化坐标");
        }
        return instruction.execute(point);
    }

    public Point getPoint() {
        return point;
    }

    public Point run(LinkedList<Instruction> instructions) {
        if (null == point) {
            throw new RuntimeException("请初始化坐标");
        }
        instructions.forEach(instruction -> point = instruction.execute(point));
        return getPoint();
    }
}

File: mars-java-kotlin/src/main/kotlin/Rover.kt
import java.util.*

class Rover {

    private lateinit var point: Point

    fun getPoint(): Point {
        return point
    }

    fun deploy(point: Point) {
        this.point = point
    }

    fun run(instructions: LinkedList<Instruction>): Point {
        instructions.forEach { point = it.excute(point) }
        return point
    }

    fun run(instruction: Instruction): Point {
        return instruction.excute(point)
    }

}